apps/lib/apps.c

```diff
@@ -3258,3 +3259,35 @@ void app_params_free(OSSL_PARAM *params)
         OPENSSL_free(params);
     }
 }
+
+EVP_PKEY *app_keygen(EVP_PKEY_CTX *ctx, const char *alg, int bits)
+{
+    EVP_PKEY *res = NULL;
+
+    if (alg != NULL) {
+        if (bits > 0)
+            BIO_printf(bio_err, "Generating a %s key, with %d bits\n",
+                       alg, bits);
+        else
+            BIO_printf(bio_err, "Generating a %s key\n", alg);
+    }
+    if (!RAND_status())
+        BIO_printf(bio_err, "Warning: generating random key material may take a long time\n"
+                   "if the system has a poor entropy source\n");
+    if (EVP_PKEY_keygen(ctx, &res) <= 0)
+        app_bail_out("%s: Error generating %s key\n", opt_getprog(), alg);
```

Comment:
This does not handle well the case where alg == NULL. Perhaps it would be better to use an additional argument instead of alg == NULL to indicate the verbosity? IMO the verbose message `Generating a .... key ...` should not be added for existing cases where there was no verbose message printed now.

Reply:
Good points. Done.
Adding a `verbose` parameter makes things cleaner. 
Made sure that also `app_paramgen()` handles the case `alg == NULL` well.